tests/filter.json

```diff
@@ -466,6 +466,45 @@
         {"d": "f"}
       ]
     },
+    {
+      "name": "non-singluar existence, wildcard",
+      "selector" : "$[?@.*]",
+      "document" : [1, [], [2], {}, {"a": 3}],
+      "result": [
+        [2],
+        {"a": 3}
+      ]
+    },
+    {
+      "name": "non-singluar existence, multiple",
+      "selector" : "$[?@[0, 0, 'a']]",
+      "document" : [1, [], [2], [2, 3], {"a": 3}, {"a": 3, "b": 4}],
```

Comment:
This could use an object that doesn't include an `a` property.  (Would not appear in the result set.)